test/FunctionalTests/Microsoft.OData.Core.Tests/ScenarioTests/UriParser/SetBasedOperationTests.cs

```diff
@@ -0,0 +1,420 @@
+                {
+                });
+            parse.ShouldThrow<ODataException>().WithMessage(
+                ODataErrorStrings.RequestUriProcessor_CannotApplyEachOnSingleEntities("People"));
+        }
+
+        [Fact]
+        public void SetBasedOperations_EachSegmentOnEntityCollection_ReturnsSuccess()
+        {
+            ParseUriAndVerify(
+                new Uri("http://gobbledygook/People/$each"),
+                (oDataPath, filterClause, aliasNodes) =>
+                {
+                    oDataPath.Count.Should().Be(2);
+
+                    EachSegment eachSegment = oDataPath.Last() as EachSegment;
+                    eachSegment.Should().NotBeNull();
+                    eachSegment.TargetEdmNavigationSource.Should().Be(HardCodedTestModel.GetPeopleSet());
+                });
+        }
+
+        [Fact]
+        public void SetBasedOperations_EachSegmentOnEntityCollectionWithFilterQuery_ReturnsSuccess()
+        {
+            ParseUriAndVerify(
+                new Uri("http://gobbledygook/People/$each?$filter=ID eq 42"),
+                (oDataPath, filterClause, aliasNodes) =>
+                {
+                    oDataPath.Count.Should().Be(2);
+
+                    filterClause.Expression.ShouldBeBinaryOperatorNode(BinaryOperatorKind.Equal).And.Right.ShouldBeConstantQueryNode(42);
+
+                    EachSegment eachSegment = oDataPath.Last() as EachSegment;
+                    eachSegment.Should().NotBeNull();
+                    eachSegment.TargetEdmNavigationSource.Should().Be(HardCodedTestModel.GetPeopleSet());
+                });
+        }
+
+        [Fact]
+        public void SetBasedOperations_EachSegmentOnEntityCollectionWithFilterSegment_ReturnsSuccess()
+        {
+            ParseUriAndVerify(
+                new Uri("http://gobbledygook/People/$filter(@p1)/$each?@p1=ID eq 42"),
+                (oDataPath, filterClause, aliasNodes) =>
+                {
+                    oDataPath.Count.Should().Be(3);
+
+                    List<FilterSegment> filterSegments = oDataPath.OfType<FilterSegment>().ToList();
+                    filterSegments.Count.Should().Be(1);
+
+                    filterSegments[0].Expression.ShouldBeParameterAliasNode("@p1", EdmCoreModel.Instance.GetBoolean(false));
+                    filterSegments[0].TargetEdmType.ToString().ShouldBeEquivalentTo(HardCodedTestModel.GetPersonType().ToString());
+                    filterSegments[0].SingleResult.Should().BeFalse();
+
+                    filterClause.Should().BeNull();
+                    aliasNodes["@p1"].ShouldBeBinaryOperatorNode(BinaryOperatorKind.Equal).And.Right.ShouldBeConstantQueryNode(42);
+
+                    EachSegment eachSegment = oDataPath.Last() as EachSegment;
+                    eachSegment.Should().NotBeNull();
+                    eachSegment.TargetEdmNavigationSource.Should().Be(HardCodedTestModel.GetPeopleSet());
+                });
+        }
+
+        [Fact]

```

Comment:
>[Fact] [](start = 8, length = 6)

maybe add a comment on what this means?  this would be used, i.e., in a PATCH to update all people with id eq 42, and return the updated people whose SSN was equal to 'num'

Reply:
:+1: